### PR TITLE
switched to `sphinx-design`

### LINKED ISSUE #8213: Migrat `sphinx-panels` to `sphinx-design` extension in documentation
### What should we add?

`sphinx-panels` extension is not actively maintained and the maintainer recommends migrating to `sphinx-design` extension: https://github.com/executablebooks/sphinx-panels#sphinx-panels A migration guide can be found here: https://sphinx-design.readthedocs.io/en/latest/get_started.html#migrating-from-sphinx-panels

See related discussions here: https://github.com/Qiskit/qiskit-terra/pull/8124#issuecomment-1143704773 and the comments below.

### PR DESCRIPTION
**Pull request by prakharb10**
### Summary
Closes #8213 

### Details and comments
The `dropdown` directive is used in the docs. It is carried over from `sphinx-panels` to `sphinx-design` (no migration docs).
Tested by building the docs locally - no errors and the `dropdown` docs are working as expected.

### CODE REVIEW

**Review by javabster — COMMENTED** [review]
Hi @prakharb10 thank-you for working on this! Can you share some screenshots of what the UI looks like when you run it locally so we can validate it looks as expected?

File: docs/conf.py [hunk 1]
```diff
@@ -46,7 +46,7 @@
 numfig_format = {"table": "Table %s"}
 
 # The language for content autogenerated by Sphinx or the default for gettext content translation.
-language = None
+language = "en"
```
[the remark this answers is not in the document]
Comment (prakharb10) [reply docs/conf.py:49]: Of course! We're currently building docs using the following command:

https://github.com/Qiskit/qiskit-terra/blob/8d5e3ca6765a7c5ec6b8c92f304f50ed01dbc8eb/tox.ini#L77

The `-W` flag here turns warnings into errors as outlined in sphinx docs [here](https://www.sphinx-doc.org/en/master/man/sphinx-build.html#cmdoption-sphinx-build-W). 
Not setting a language code, i.e., `language = None` raised a [Warning ](https://dev.azure.com/qiskit-ci/qiskit-terra/_build/results?buildId=37390&view=logs&j=abfb3e93-d1ff-5b1b-cb4c-74d6dcf536f9&t=60edc4b3-4259-59c8-ae13-dc31febe2d3d&l=21) and resulted in tests failing on the previous commit.

Therefore, to build the docs successfully, we either need to let go of the `-W` arg or set a language as above. My understanding led me to opt for the latter.
I hope this clears things up.

P.S. I know docs were being built successfully before, even with warnings and the `-W` flag. My best guess for this being triggered now is that `sphinx-panels` bounded `sphinx` to `<5.0.x`. After migration to `sphinx-design`, the `sphinx` version resolved to [`5.0.2`](https://dev.azure.com/qiskit-ci/qiskit-terra/_build/results?buildId=37402&view=logs&j=abfb3e93-d1ff-5b1b-cb4c-74d6dcf536f9&t=60edc4b3-4259-59c8-ae13-dc31febe2d3d&l=20) and the `-W` arg wasn't being honored on earlier versions.

**Review by jakelishman — APPROVED** [review]
Looks good to me, thanks.  We'll need to update the metapackage docs build (qiskit/qiskit) accordingly as well, but that's all fine.

File: docs/conf.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (jakelishman) [reply docs/conf.py:49]: Yeah, Sphinx 5 made `language = None` a warning, we treat warnings as errors, and `sphinx-panels` was limiting Sphinx to pre-5.  I've made a similar change on Aer already, and we can update the metapackage when we update it to account for the changes here.

**Pull request merged**